_drafts/A BLE Primer.md

```diff
@@ -0,0 +1,1165 @@
+
+- Connect the development kit to your computer (via the micro-USB to USB-A cable)
+- Make sure the development kit jumpers are in the right settings:
+	- **SW6** set to "Default" (**not** "nRF ONLY")
+	- **SW9** (nRF power source) set to "VDD"
+	- **Power** set to "ON"
+	
+	<br/>
+	![](img/ble-primer/nRF52840_dev_kit.png)
+	
+- Erase the development kit by running the following command. This will erase the chipset on the board, just to make sure we are starting from a clean slate.
+
+	```terminal
+	$ nrfjprog -f NRF52 --eraseall
+	```
+
+- Next, we will flash the chipset with Nordic's SoftDevice which acts as both the BLE stack and RTOS (Real-Time Operating System) for the nRF52 chipset. This is crucial before flashing our application (since the binaries compile separately).
+
+	To do this, run the following command:
+
+	```terminal
+	$ nrfjprog -f NRF52 --program <nRF5 SDK Folder>/components/softdevice/s140/hex/s140_nrf52_6.1.1_softdevice.hex --chiperase --verify
+	```
+	
+	Don't forget to change the `<nRF5 SDK Folder>` portion to match the location where you installed/unzipped the nRF5 SDK.
+	
+- The final step is to flash the application binary to the chipset. To do this, run the following command:
+
+	```terminal
+	$ nrfjprog -f NRF52 --program _build/nrf52840_xxaa.hex --verify
+	```
+	
+That's it! Our application should now be running on the board. To verify that, we will be using a serial terminal program.
+
+In the next section, we'll go through setting up the terminal program and then verify that the application is running correctly.
+
+#### 3. Serial Terminal Program Setup
+As we mentioned earlier, we'll be using a program called CoolTerm.
+
+There are three parts to get this working:
+
+- Run the CoolTerm application
+- Make sure the serial port settings are correct (listed at [this link](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.3.0/uart_example.html)):
+ 
+	![](img/ble-primer/CoolTerm_Settings.png)
+	
+	Now, hit **OK**. 
+
+- Finally, connect to the serial port by clicking the "Connect" button:
+	
+	![](img/ble-primer/CoolTerm_Connect.png)
+	
+	You may not see any output since the program probably started before you connected. To reset the development board, we can simply run the following command from the Terminal:
+	
+	```terminal
+	$ nrfjprog -f NRF52 --reset
+	```
+	
+The first time you run the application you'll most likely get an error that looks something like this:
+
+![](img/ble-primer/ram_offset_size_error.png)
+
+This is due to the fact that we modified the value of the macro for the number of vendor-specific UUIDs in the SoftDevice. The good part is that the log output tells us exactly which values we need to use for RAM offset and RAM size to fix this problem.
+
+In order to apply the fix, we need to edit the file named `ble_app_template_gcc_nrf52.ld` located at `ble_app_template/pca10056/s140/armgcc/` and modify the values of `  RAM (rwx) :  ORIGIN` and `  RAM (rwx) :  LENGTH`:
```

Comment:
this doesn't seem to have formatted correctly ... I think it's due to the space before the ` ...
![image](https://user-images.githubusercontent.com/7494333/62084730-f1b2b400-b227-11e9-878e-1c5bde840dbd.png)

Reply:
Fixed